### ISSUE TITLE
Team-Subdomains try to get their cookie from Topdomain

### ISSUE THREAD

**Issue by LarsFronius** [ether/pad#229]
We want to do following:
We currently have Etherpad in our scientific institute. And we don't want to make it public cause of spam in public pads and so on.
What I want to do is making some team-sites public.
So what I tried to do is: domain.xyz has ip 1.1.1.1 and is blocked from our firewall.
publicteam.domain.xyz has ip 1.1.1.2 and has http/https open in the firewall.

It is no problem now to get to publicteam.domain.xyz, cause the http and https ports are open and dns points to 1.1.1.2.
Here is the problem: When I try to login on publicteam.domain.xyz from outside of the institute, etherpad leads me to domain.xyz/setCookie (and so on), which is blocked and we do not want to unblock.
Is there a way to get the Cookie from publicteam.domain.xyz? And above all: Shouldn't this be the default? Because I do not want to go to domain.xyz, but to my teamsite.

**Comment by sesam** [comment]
With your current proposed setup, even if it would work as you proposed, anyone could add in their local computer's HOSTS-file (etc/hosts) and if it is the same etherpad server that answers on both IP:s then I'd trivially get around your DNS-level "blocking".
Probably you instead need to put the blocking in a separate proxying webserver which sits in front of etherpad. I hear lighty and nginx are great at such work.

If you have an idea how to rewrite etherpad to not use the "home" domain while logging in, please make a new ticket and try to point to where in the code. From what I understand though, cookies need to be set on the base domain name in order to work on subdomains. Etherpad was probably designed in the typical standard fashion to not have separate cookies for each domain. And please be careful with the terms, a "topdomain" or TLD is .com and .co.uk :-)

**Comment by redhog** [comment]
I think the right solution is to use one IP, and use the HTTP_AUTH/ip-based allow/deny functions of Apache (or nginx) to control access to different subdomains and/or paths.

This way, the setup would mimic a fully public site closely, just not allow access to all parts, which should greatly simplify things for you.
